service-search-nls/src/main/java/fi/nls/oskari/control/view/modifier/param/AddressParamHandler.java

```diff
@@ -64,17 +73,16 @@ public boolean handleParam(final ModifierParams params) throws ModifierException
     protected ArrayList<String[]> getCoordinatesFromAddress(
                               String searchString, Locale locale) {
 
-        final ArrayList<String[]> lat_lon = new ArrayList<String[]>();
+        final ArrayList<String[]> lat_lon = new ArrayList<>();
 
         final SearchCriteria sc = new SearchCriteria();
-        sc.addChannel(MaastoAddressChannelSearchService.ID);
-        //searchString = URLDecoder.decode(searchString, "UTF-8");
+        sc.addChannel(channelID);
         sc.setSearchString(searchString);
         sc.setLocale(locale.getLanguage());
 
         try {
             final Query query = searchService.doSearch(sc);
-            final ChannelSearchResult result = query.findResult(MaastoAddressChannelSearchService.ID);
+            final ChannelSearchResult result = query.findResult(channelID);
             for(SearchResultItem item : result.getSearchResultItems()) {
                 lat_lon.add(item.getContentURL().split("_"));
```

Comment:
This probably needs changing also (as NLSFIGeocodingSearchChannel doesn't set contentURL for SearchResultItems), ie. something like:

lat_lon.add(new String[] { String.valueOf(item.getLat()), String.valueOf(item.getLon()) });

Not sure about axis ordering though..